History page

Nawsen opened the pull request:
This is just a basic styling idea, please give feedback so I can improve this.
I added the first 4 events just as a demo, if ppl like it I will add them all.

Desktop
![history](https://user-images.githubusercontent.com/9265860/34122986-6ffa7d4e-e42e-11e7-9b0d-d6e47df8c2ae.png)
Mobile
![history2](https://user-images.githubusercontent.com/9265860/34123005-7a0d4938-e42e-11e7-9f6d-816ddb3ecd21.png)

Mobile still has some layout issues I'm working on.
If you have different svg's/icons you would like me to use, other colors or anything else? Let me know ;)

Commit:
Added history page with basic styling

Nawsen commented:
#38

NxtChg commented:
Oy, that's awesome! :)

The history list is being filled, so if you can then reformat it with this style - that's a $50 bounty from me (and I hope others would join too).

One thing I would suggest is to reclaim some vertical space by removing 'read more' and instead adding some small superscript link at the end of the text, something like "[proof]" or even "[*]". The list will be long, so it makes sense to keep the height to the minimum.

NxtChg commented:
Actually, I think I can write a script to produce the raw structure. But the icons still need to be tweaked by hand...

Nawsen commented:
> One thing I would suggest is to reclaim some vertical space by removing 'read more' and instead adding some small superscript link at the end of the text, something like "[proof]" or even "[*]". The list will be long, so it makes sense to keep the height to the minimum.

I was also not really happy with the "Read more" link. I didn't think of a superscript link, will try that out.

> Actually, I think I can write a script to produce the raw structure. But the icons still need to be tweaked by hand...

Writing some javascript to generate the structure is not a problem for me either.
Maybe its a good idea to keep the data (text, link, icon, color, ...) in a json file?

NxtChg commented:
> Maybe its a good idea to keep the data (text, link, icon, color, ...) in a json file?

Thought of that too. This can allow re-sorting by JS later. But I am worried that performance might suffer, if we create the structure on page load...

Or did you mean for internal usage, for example, to generate HTML from it? This might be a good idea too.

Nawsen commented:
> Thought of that too. This can allow re-sorting by JS later. But I am worried that performance might suffer, if we create the structure on page load...

I don't think this will be a huge performance issue atm. However the list keeps getting bigger every day so you might be right...

> Or did you mean for internal usage, for example, to generate HTML from it? This might be a good idea too.

This would be better performance wise but we would lose the sorting feature, or at least make it much harder (and uglier).

NxtChg commented:
Well, we can store the list in json for now, generate a static html from it by a tool, and then later it will be easy to make a dynamic page out of it.

Commit:
Improved history page layout

Nawsen commented:
Some progress with the remarks of @NxtChg 

Desktop
![image](https://user-images.githubusercontent.com/9265860/34126733-e8d08c5c-e43a-11e7-9549-3243649a19ae.png)

Mobile
![image](https://user-images.githubusercontent.com/9265860/34126693-c7436c26-e43a-11e7-85ab-4bcaab7c9f2f.png)

Let me know what you think!

NxtChg commented:
Perfect.

NxtChg commented:
Minor: maybe in the mobile version make the date lighter or add ':' after it to help separate it visually from the text.

Also the triangle breaks shadow :)

Commit:
History page: Made date lighter and added some more padding

Commit:
Merge branch 'master' of github.com:Nawsen/bitcoincashorg-website into history

NxtChg commented:
#38 updated.

Commit:
Merge branch 'master' of github.com:bitcoincashorg/website into history

Commit:
History page: Fixed shadow issue

Commit:
History page: Timeline build from json

Nawsen commented:
New demo [here](https://nawsen.github.io/bitcoincashorg-website/html/history.html)

- Fixed triangle shadow
- Mobile version date is lighter and has some more padding
- HTML is generated from the json file

zquestz commented:
This is fantastic @Nawsen. I think this is very close to merging. One last thing I think we should tweak are the icon colors. I think we can come up with shades that match the layout a bit better. Other than that I think we are GTG!

zquestz commented on html/css/custom.css:
```diff
@@ -137,4 +137,246 @@ p.question {
\ No newline at end of file
+}
+
+.bcc-timeline-container {
```
NIT: We should update these selectors to use bch =)

zquestz commented:
We should also probably add a `History` link under community. Also think we should make the link `bitcoincash.org/history/` instead of just having a history.html file. What do you think?

Nawsen commented:
> One last thing I think we should tweak are the icon colors. I think we can come up with shades that match the layout a bit better.

I will do some more tinkering with this.

> We should also probably add a History link under community.

Will do.

> Also think we should make the link bitcoincash.org/history/ instead of just having a history.html file. What do you think?

I think we should have a history.html.erb file. As the history page will also have a header/footer I assume this also has to be translated? But I agree with you that the url should be /history. I think this should be done with some rewrite rules on the server side?

Commit:
Merge branch 'master' of https://github.com/bitcoincashorg/website into history

Commit:
Updated bcc selectors to bch

Commit:
added history link to index.html

Commit:
Merge branch 'master' of https://github.com/bitcoincashorg/website into history

Commit:
History html generated by Rakefile

Commit:
created correct folders on build

Commit:
Finished build & bugfixes

Commit:
Merge branch 'master' of https://github.com/bitcoincashorg/website into history

Commit:
wrong file edit reverted

Commit:
Changed colors for history items

NxtChg commented:
@Nawsen Sorry, I am not contributing anything to this project until the owner comes out of the woods and answers #96.

zquestz commented:
Taking a look now. Thanks so much for doing this @Nawsen =)

zquestz commented:
OK, loaded it up. Overall this looks great, really appreciate the work you have done so far!

A few things I would still clean up:
- The page probably needs a header that says "History". Not sure if this should be just a header tag or a header image.
- I still think the colors of the icons could match better with the orange. I can play with this a bit if you want.
- Should we update the news to include January so everything is current when this goes live?

Lets try to take this over the finish line. =)

Nawsen commented:
> A few things I would still clean up:
> 
> The page probably needs a header that says "History". Not sure if this should be just a header tag or a header image.
Will do that.
> I still think the colors of the icons could match better with the orange. I can play with this a bit if you want.
Yes, that would be greatly appreciated!
> Should we update the news to include January so everything is current when this goes live?
I think so, however I have no list of all those events. Do you have something?
> The top entry should not have a line coming off the top.
Will fix that.

schancel commented:
Since @NxtChg has reneged on his bounty, I will provide the bounty and double it to $100 USD.  Although, I don't think this comes close to representing the fair market value of this work.  Thank you for all the effort @Nawsen .

@Nawsen, please provide me an address to send some BCH to.

Commit:
Added history Header & Removed top line from timeline

Commit:
Added generated history files to gitIgnore

Nawsen commented:
I added a header to the history page. This is just a h3 like the other headers.
I removed the top line coming of the top entry.

Things I would like some help with: @schancel @zquestz 
- Better colors for each type of event
- More events after 18 November 2017


> @Nawsen, please provide me an address to send some BCH to.

1NU3zisaw7SP5hfYgDQ7BwoE9kxGW5yMzv

schancel commented:
Thank you Nawsen.  You should have received some BCH.

schancel commented:
Hello,

Thank you for your contribution.

We are migrating to a combined repository to improve the website experience.  This includes:

* Adding a developer portal, which includes protocol specification, and other resources
* Switching to a common template rendering sysmte ([Jekyll](https://jekyllrb.com/))
* Automated deployments to a staging area to facilitate in interative development.

Please help us by submitting your pull request to the new repository:

http://github.com/bitcoincashorg/bitcoincash.org
